.github/workflows/codegen.yml

```diff
@@ -0,0 +1,42 @@
+# All code generation should be run prior to pull request. Running it again should not produce a diff.
+name: "Codegen Verifier"
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'ccip-develop'
+
+jobs:
+  codegen-verifier:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        go-version: ['1.21']
+    defaults:
+      run:
+        working-directory: .
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Setup Go ${{ matrix.go-version }}
```

Comment:
Should we also remove all the generated file committed to make sure we have a diff? It might be useful to detect files to clean up.

Reply:
https://github.com/smartcontractkit/ccip/blob/451984ad0469c7d685e305dba0a0d94eb0c9a053/.github/workflows/ci-core.yml#L426

Reply:
Done. :+1: